.github/workflows/ci.yml

```diff
@@ -642,6 +642,44 @@ jobs:
           path: diagnostics*.gz
           retention-days: 1
 
+  unit-benchmark:
+    name: Unit Benchmarks
+    runs-on: ubuntu-latest
+    needs: check
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
+
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly
+          profile: minimal
+
+      - name: Setup Cache for Rust and Benchmarks
+        uses: Swatinem/rust-cache@v2
+        with:
+          # Additional non workspace directories, separated by newlines
+          cache-directories: ./benches
+          key: ${{ runner.os }}-benchmark
+
+      - name: Run Benchmarks
+        run: cargo +nightly bench | tee benchmarks.txt
+
+      - name: Store benchmark result
+        uses: benchmark-action/github-action-benchmark@v1
+        with:
+          # What benchmark tool the output.txt came from
+          tool: 'cargo'
+          # Where the output from the benchmark tool is stored
+          output-file-path: benchmarks.txt
+          # GitHub API token to make a commit comment
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          # Enable alert commit comment
+          comment-on-alert: true
+          # Where the previous data file is stored
+           external-data-json-path: ./benches/benchmark-data.json
```

Comment:
looks like there is indentation issue here

Reply:
YML is picky.   @EstebanBorai If you have VSCode or other tools, there is YAML plug-in

Reply:
Fixed! Yeah I have to check for linters on this

Reply:
I'm rendering spaces already, but sometimes it is a bit tricky to see